client/lib/gsuite/index.js

```diff
@@ -100,12 +100,24 @@ function getGSuiteSupportedDomains( domains ) {
 			includes( [ domainTypes.REGISTERED ], domain.type ) &&
 			( domain.hasWpcomNameservers || hasGSuite( domain ) );
 		const mapped = includes( [ domainTypes.MAPPED ], domain.type );
-		const notOtherProvidor =
+		const notOtherProvider =
 			domain.googleAppsSubscription && domain.googleAppsSubscription.status !== 'other_provider';
-		return ( wpcomHosted || mapped ) && canDomainAddGSuite( domain.name ) && notOtherProvidor;
+		return ( wpcomHosted || mapped ) && canDomainAddGSuite( domain.name ) && notOtherProvider;
 	} );
 }
 
+/**
+ * Returns the primary domain name if the domain name supports GSuite
+ *
+ * @param {Array} domains - list of domain objects
+ * @returns {String} - name of the domain if valid
+ */
+function getGSuiteSupportedPrimaryDomainName( domains ) {
+	const eligibleGSuiteDomains = getGSuiteSupportedDomains( domains );
+	const selectedGSuiteDomain = find( eligibleGSuiteDomains, 'isPrimary' );
+	return get( selectedGSuiteDomain, 'name' );
+}
```

Comment:
The legacy code is a bit difficult to navigate, because it uses very similar semantic such as `supported` and `eligible`. What is a supported domain? What is an eligible domain? It's very difficult to understand the difference, especially since it seems it depends on the context. I'm concerned that adding this `getGSuiteSupportedPrimaryDomainName()` function is not going to make this clearer, especially given how similar it is with `getEligibleGSuiteDomain()`.  Maybe we could simply reuse the latter, and just update it to return the primary domain (or the first domain if no primary domain is set). I'll have a look.

Reply:
> We should probably handle the case where a user has set their primary domain to their default *.wordpress.com address. In that particular case, we could still display the nudge for the first domain that is eligible to G Suite.

I've tried to implement this, but I'm hitting a wall because of the way our code is designed. The list of supported domains doesn't contain the default `*.wordpress.com` address so it's difficult to know whether the user indeed set it as primary, or used another domain as primary which is not eligible to G Suite (such as a mapped domain that doesn't have WordPress.com nameservers). Let's drop my initial idea, and only display the nudge for the primary domain.